plugins/woocommerce/includes/wc-template-functions.php

```diff
@@ -1402,16 +1402,16 @@ function woocommerce_show_product_loop_sale_flash() {
 	 * Get the product thumbnail, or the placeholder if not set.
 	 *
 	 * @param string $size (default: 'woocommerce_thumbnail').
-	 * @param int    $deprecated1 Deprecated since WooCommerce 2.0 (default: 0).
-	 * @param int    $deprecated2 Deprecated since WooCommerce 2.0 (default: 0).
+	 * @param  array $attr Image attributes.
+	 * @param  bool  $placeholder True to return $placeholder if no image is found, or false to return an empty string.
 	 * @return string
 	 */
-	function woocommerce_get_product_thumbnail( $size = 'woocommerce_thumbnail', $deprecated1 = 0, $deprecated2 = 0 ) {
+	function woocommerce_get_product_thumbnail( $size = 'woocommerce_thumbnail', $attr = array(), $placeholder = true ) {
```

Comment:
To avoid potential type errors, let's add some defensive checks on the `$attr` and `$placeholder` parameters.

Something like:
```php
if ( ! is_array( $attr ) ) {
    $attr = array();
}

if ( ! is_bool( $placeholder ) ) {
    $placeholder = true;
}
```

Reply:
You're right. Thanks!